shared/common-adapters/text.desktop.js

```diff
@@ -81,313 +25,47 @@ class Text extends Component<void, Props, void> {
-      'BodySmallSemibold': styles.textBodySmallSemibold,
-      'BodyXSmall': styles.textBodyXSmall,
-      'BodyXSmallLink': styles.textBodyXSmallLink,
-      'Error': styles.textError,
-      'Header': styles.textHeader,
-      'HeaderBig': styles.textHeaderBig,
-      'HeaderError': styles.textHeaderError,
-      'HeaderJumbo': styles.textHeaderJumbo,
-      'HeaderLink': styles.textHeaderLink,
-      'InputHeader': styles.textInputHeader,
-      'Terminal': {...styles.textTerminal, color: (this.context.inTerminal ? globalColors.blue3 : globalColors.darkBlue)},
-      'TerminalCommand': styles.textTerminalCommand,
-      'TerminalComment': styles.textTerminalComment,
-      'TerminalEmpty': styles.textTerminalEmpty,
-      'TerminalPrivate': styles.textTerminalPrivate,
-      'TerminalPublic': styles.textTerminalPublic,
-      'TerminalSmall': {...styles.textTerminalSmall, color: (this.context.inTerminal ? globalColors.blue3 : globalColors.darkBlue)},
-      'TerminalUsername': styles.textTerminalUsername,
-    }[this.props.type]
-  }
-
   render () {
-    const typeStyle = this._typeStyle()
-    let inline = true
-    if (this.props.hasOwnProperty('inline')) {
-      inline = this.props.inline
-    }
-
-    // $FlowIssue not types in the dict
-    const linkClassname = LinkTypes[this.props.type] ? 'hover-underline' : null
-
     const style = {
-      ...typeStyle,
-      ...(LinkTypes[this.props.type] ? {cursor: 'pointer'} : {}),
-      ...Text._colorStyleBackgroundMode(this.props.backgroundMode || 'Normal', this.props.type),
-      ...(this.props.lineClamp ? lineClamp(this.props.lineClamp) : {}),
-      ...(this.props.onClick ? globalStyles.clickable : {}),
-      ...(inline ? {...Text._inlineStyle(this.props.type, this.context)} : {display: 'block'}),
+      ...getStyle(this.props.type, this.props.backgroundMode, this.props.lineClamp, !!this.props.onClick),
       ...this.props.style,
     }
 
-    const terminalPrefix = this._terminalPrefix(this.props.type)
-    const className = (this.props.className || '') + ' ' + (linkClassname || '')
+    const meta = metaData[this.props.type]
 
-    if (this.props.contentEditable) {
-      return (
-        <span
-          ref='text'
-          className={className}
-          style={style}
-          contentEditable={true}
-          onKeyUp={this.props.onKeyUp}
-          onKeyDown={this.props.onKeyDown}
-          onClick={this.props.onClick} />)
-    }
+    const className = [
+      this.props.className,
+      meta.isLink ? 'hover-underline' : null,
+    ].filter(Boolean).join(' ')
 
-    return (
-      <span
-        ref='text'
-        className={className}
-        style={style}
-        onClick={this.props.onClick}>{terminalPrefix}{this.props.children}</span>)
+    return <span
+      ref={ref => { this._span = ref }}
+      className={className}
+      style={style}
+      onClick={this.props.onClick}>{this.props.children}</span>
   }
 }
 
-Text.contextTypes = {
```

Comment:
Note: we don't use context anymore. Its an experimental feature and this is our only use of it and i think its a little magical so let's just be explicit.